### ISSUE TITLE
Inconsistent row selection behaviour

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#15692]
**Describe the bug**
When I use the keyboard to select one row of text the behaviour in Gutenberg is inconsistent:

- if the row is NOT the last row, I can press SHIFT + Down Arrow and it will select every letter up to the first line break
- if the row is the last then pressing SHIFT + Down Arrow I will unwillingly enter block selection bode and will then end up selecting two entire blocks. Exiting this mode is highly unintuitive and without using the mouse I ended up deleting content.

**Expected behavior**
I expect consistent row selection, and if SHIFT + Down Arrow is intended to be used for block selection then I should have to press it once more after the last row is selected, to signify that my clear intent is to select beyond the current block.

**Screenshots**

![row-selection](https://user-images.githubusercontent.com/107534/57909787-6793ba00-788c-11e9-9d32-b2f6de5e45ae.gif)

**Comment by afercia** [comment]
For a similar, previous, issue see #12322 / #14448.

Text selection with keyboard within a block should replicate the native behavior in a textarea.

The last press of Shift + Down Arrow should select text within a block till the end of the block text. 

Similarly when selecting in the opposite direction pressing Shift + Up Arrow.

**Comment by draganescu** [comment]
@ellatrix as Andrea mentioned above I looked at all the other issues but in all fairness it's a bit hard to follow, so the question is: 

Is this issue I described here fixed "as good as possible" or, if not, what is the current situation, because all the other issues ( #12322 / #14448  etc. )  are closed as fixed but this still happens.

**Comment by draganescu** [comment]
This seems to be solved in current master.

![arrow-selection](https://user-images.githubusercontent.com/107534/77777683-935d0000-7058-11ea-8222-1713daf9c294.gif)

It does look like SHIFT + arrow down works but if you then press SHIFT + arrow up you switch to block selection unexpectedly. But that is another issue, I will close this one.
